api_app/analyzers_manager/utils.py

```diff
@@ -65,16 +67,17 @@ def set_report_and_cleanup(analyzer_name, job_id, report):
         job_object.save(update_fields=["finished_analysis_time"])
 
 
-def get_basic_report_template(analyzer_name):
-    return {
-        "name": analyzer_name,
-        "success": False,
-        "report": {},
-        "errors": [],
-        "process_time": 0,
-        "started_time": time.time(),
-        "started_time_str": timezone.now().strftime("%Y-%m-%d %H:%M:%S"),
-    }
+def get_report_model_object():
+    report_obj = AnalyzerReport(
+        success=False,
+        report={},
+        errors=[],
+        start_time=datetime.now(),
+        end_time=datetime.now(),
+    )
```

Comment:
I assume, I will also need to pass in `Analyzer` and `Job` model objects (since `AnalyzerModel` has a foreign key relationship with them? If so, for `Job`, i'll need to do some refactoring but i was confused how to incorporate the `Analyzer` model here.

Reply:
This function shouldn't be inside the `utils.py`. Remember that we are refactoring the codebase to make it better so we don't necessarily need to adopt previous code. 

A better approach inline with new changes would be to have a member method `init_report` inside `Analyzer` model. The signature being something like: `def init_report(self) -> AnalyzerReport`

Reply:
Another way could be to have this method inside `Job` model. `def init_report(self, analyzer: Analyzer) -> AnalyzerReport`. I'll let you decide which one is better as you are coding it.

Reply:
I put it in Job cause the report model as i understood, represents a Job's report (which may contain individual reports from different analyzers) so it made more sense to me that way